src/components/dagshub/data-engine/queryBuilder/QueryBuilderContext.tsx

```diff
@@ -227,14 +221,24 @@ export const QueryBuilderProvider = ({
     return condition;
   };
 
+  const debouncedOnChange = useCallback(
```

Comment:
Wouldn't be nicer to just debounce `onChange` in the `ConditionTextField` ? That way only the text keyboard inputs are debounced.